whois-nrtm4/src/main/java/net/ripe/db/nrtm4/dao/NrtmVersionInfoRepository.java

```diff
@@ -43,40 +43,40 @@ public class NrtmVersionInfoRepository {
     };
     private final RowMapper<NrtmVersionInfo> rowMapper = rowMapperWithOffset.apply(0);
     private final JdbcTemplate jdbcTemplate;
-    private final DateTimeProvider dateTimeProvider;
 
-    public NrtmVersionInfoRepository(
-        @Qualifier("nrtmDataSource") final DataSource dataSource,
-        final DateTimeProvider dateTimeProvider
-    ) {
+    public NrtmVersionInfoRepository(@Qualifier("nrtmDataSource") final DataSource dataSource) {
         this.jdbcTemplate = new JdbcTemplate(dataSource);
-        this.dateTimeProvider = dateTimeProvider;
     }
 
     public List<NrtmVersionInfo> findLastVersionPerSource() {
```

Comment:
Here we could filter by type, right? passing the type as parameter. Or do we need all of types always?

Reply:
Yes, Its the lastVersionPerSource not perSourcePerType. We need it to decide the version